packages/lit-dev-content/site/tutorials/content/tooltip/07.md

```diff
@@ -0,0 +1,105 @@
+element and a callback that will render tooltip content on demand when
+its needed.
+
+Add a `static` method to the `SimpleTooltip` class called
+`lazy`. It will setup event listeners on the target element that listen on the
+same events used to trigger the tooltip. The listener creates the tooltip
+element and calls the user callback with it to fill in the content before
+showing it. Near the top of the class definition, add:
+
+### simple-tooltip.ts
+{% switchable-sample %}
+
+```ts
+static lazy(target: Element, callback: (target: SimpleTooltip) => void) {
+  let called = false;
+  enterEvents.forEach(name => target.addEventListener(name, () => {
+    if (!called) {
+      called = true;
+      const tooltip = document.createElement('simple-tooltip');
+      callback(tooltip);
+      target.parentNode!.insertBefore(tooltip, target.nextSibling);
+      tooltip.show();
+    }
+  }, {once: true}));
+}
+```
+
+```js
+static lazy(target, callback) {
+  let called = false;
+  enterEvents.forEach((name) => target.addEventListener(name, () => {
+    if (!called) {
+      called = true;
+      const tooltip = document.createElement('simple-tooltip');
+      callback(tooltip);
+      target.parentNode.insertBefore(tooltip, target.nextSibling);
+      tooltip.show();
+    }
+  }, {once: true}));
+}
+```
+
+{% endswitchable-sample %}
+
+<aside class="info">The <code>once</code> option is used to automatically
+remove the event listener after it fires once. Since the callback should be
+called only once and there are multiple triggering events, a flag is used to
+manage the state.</aside>
+
+Now open the `my-content.ts` module and change the `greeting` tooltip to use
+the `lazy` method. To do so, first remove the inline tooltip:
+
+### my-content.ts
+```html
+<p>
+  <span id="greeting">Hello, ${this.name}! <span class="icon">info_outline</span></span>
+</p>
+```
+
+Then define a `firstUpdated` lifecycle callback to setup the call to `lazy`.
+
+{% switchable-sample %}
+
+```ts
+import {SimpleTooltip} from './simple-tooltip.js';
+
+@customElement('my-content')
+export class MyContent extends LitElement {
+  // ...
+  firstUpdated() {
+    const greeting = this.shadowRoot!.getElementById('greeting')!;
+    SimpleTooltip.lazy(greeting, (tooltip: SimpleTooltip) => {
+      tooltip.textContent = `${this.name}, there's coffee available in the lounge.`;
+    });
+  }
+```
+
+```js
+import {SimpleTooltip} from './simple-tooltip.js';
+
+export class MyContent extends LitElement {
+  // ...
+  firstUpdated() {
+    const greeting = this.shadowRoot.getElementById('greeting');
+    SimpleTooltip.lazy(greeting, (tooltip) => {
+      tooltip.textContent = `${this.name}, there's coffee available in the lounge.`;
+    });
+  }
+```
+
+{% endswitchable-sample %}
+
+The tooltip for the greeting element should work the same now, but it's created
+on demand when one of the "enter" events is triggered. If you do this for
```

Comment:
If you do this for all => kind of suggests you _should_, and maybe that's fine... But given that the next step is going to replace all of this code, maybe an alternate cue like,

You could do this for all of the tooltips to ensure they don't impost a performance penalty on initial page render. But adding them this way is a lot of extra code. In the next step, you'll simplify the process of adding a tooltip. (Or: remove some of this boilerplate.)